src/buffer_tree.cpp

```diff
@@ -47,13 +48,16 @@ nodes, int workers, bool reset=false) : dir(dir), M(size), B(b), N(nodes) {
 	buffer_size     = M; // probably figure out a better solution than this
 	max_buffer_size = 2 * M;
 	backing_EOF     = 0;
+	leaf_size       = floor(24 * pow(log(N) / log(2), 3)); // size of leaf proportional to size of sketch
+	leaf_size       = (leaf_size < page_size)? page_size : leaf_size; //enforce size of at least page_size
 
 	// malloc the memory for the root node
-	root_node = (char *) malloc(max_buffer_size);
+	root_node = (char *) malloc(buffer_size);
 	root_position = 0;
 
 	// malloc the memory to use for reading buffers
 	read_buffer = (char *) malloc(max_buffer_size);	
+	backup_read_buffer = (char *) malloc(2 * leaf_size);
```

Comment:
why is this a backup?

Reply:
Because it's used when we need to flush a leaf while still in the process of flushing the parent